streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -838,4 +838,28 @@ object ZSink extends ZSinkPlatformSpecificConstructors {
    */
   def sum[A](implicit A: Numeric[A]): ZSink[Any, Nothing, A, Nothing, A] =
     foldLeft(A.zero)(A.plus)
+
+  /**
+   * A sink that offers values to a given queue. Shuts down the queue if specified.
+   */
+  def toQueue[R, I](
+    queue: ZQueue[R, Nothing, Nothing, Any, Chunk[I], Any],
+    shutdown: Boolean
+  ): ZSink[R, Nothing, I, I, Unit] =
+    ZSink {
+      for {
+        state <- Ref.make[Chunk[I]](Chunk.empty).toManaged_
```

Comment:
I don't understand the semantics here. Why do we need to accumulate all chunks offered to the queue and pass them on as a leftover? This uses unbounded memory and seems pretty unsafe to me.

Beyond that, I wouldn't expect a queue sink to result in any leftovers so the effort sounds a bit redundant; we could probably use `Nothing` for leftovers here and consider anything offered to the queue to be consumed.

What do you think @luis3m?

Reply:
@iravid Sorry, I'm not familiar with ZSink 😅

So you're saying the leftover part is unnecessary? I thought of leftovers based on the comment that ZSink are composable.